onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -1202,9 +1202,12 @@ class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain,
 class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 14, float, LSTM);
 class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 14, double, LSTM);
 class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 14, MLFloat16, LSTM);
-class ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 14, 14, float, BatchNormalization);
-class ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 14, 14, double, BatchNormalization);
-class ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 14, 14, MLFloat16, BatchNormalization);
+class ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(
```

Comment:
I think the support for type `double` for the NCHW `BatchNorm` came with the very first commit. Since then the strategy has changed to be more careful in adding further type support unless warranted by a production use-case. Even though currently the CUDA NHWC support is turned OFF by default, if and when we turn it ON by default - we may have to deal with all the excess binary size growth. Maybe we should hold off on supporting the `double` type for NHWC `BatchNorm` for now if the only reason for that enhancement is to get type parity with NCHW `BatchNorm` ?

Reply:
We can add add a build flag later to enable/disable all double NHWC ops.